File: LeetCode/Easy/3483-unique-3-digit-even-numbers/3483-unique-3-digit-even-numbers.py

def genarate(vis,digits,lst,pst):
    global count
    if len(lst)==3:
        if lst[-1]%2==0 and lst[0]!=0 and lst not in pst:
            pst.append(lst.copy())
        return 
    for i in range(len(digits)):
        if vis[i]==0:
            lst.append(digits[i])
            vis[i]=1
            genarate(vis,digits,lst,pst)
            lst.pop()
            vis[i]=0
class Solution:
    def totalNumbers(self, digits: List[int]) -> int:
        vis=[0]*len(digits)
        lst=[]
        pst=[]
        genarate(vis,digits,lst,pst)
        return len(pst)
        
        
        
        
        
        

File: LeetCode/Easy/3483-unique-3-digit-even-numbers/3483-unique-3-digit-even-numbers.cpp
class Solution {
public:

    void Solve(vector<int>&digits, vector<int>&vis, string s,set<string>&st, int n){
        //step - 1 base case
        if(s.size() == 3){
            int val = s[2]-'0';
            if(val % 2 == 0) st.insert(s);
            return;
        }

        for(int i = 0 ; i < n ; i++){
            if(vis[i]==0){
                if(s.size()== 0 && digits[i]== 0){
                    continue;
                }
                s.push_back(digits[i] + '0');
                vis[i] = 1;
                Solve(digits,vis,s,st,n);
                s.pop_back();
                vis[i]=0;
            }
        }
    }
    int totalNumbers(vector<int>& digits) {
        int n = digits.size();
        vector<int>vis(n,0);

        string s;
        set<string>st;
        Solve(digits,vis,s,st,n);   
        return st.size();
    }
};